CBLClient/Apps/CBLTestServer-Java-Desktop/src/main/java/com/couchbase/mobiletestkit/javatestserver/TestServerMain.java

```diff
@@ -1,35 +1,155 @@
+
+public class TestServerMain implements Daemon {
     private static final int PORT = 8080;
     private static final String TMP_DIR = System.getProperty("java.io.tmpdir");
+    private static final String TAG = "TestServer-Java";
     private static Context context;
+    private static boolean stopped = false;
+    private static TestServerMain testserverLauncherInstance = null;
 
+    /**
+     * Main method runs as non-service mode for debugging use
+     * @param args
+     * @throws IOException
+     */
     public static void main(String[] args) throws IOException {
+        if(testserverLauncherInstance == null){
+            testserverLauncherInstance = new TestServerMain();
+        }
+
+        testserverLauncherInstance.initCouchbaseLite();
+        Log.i(TAG, "Main");
+        testserverLauncherInstance.startServer(stopped);
+    }
+
+    /**
+     * Static methods called by prunsrv to start/stop
+     * the Windows service.  Pass the argument "start"
+     * to start the service, and pass "stop" to
+     * stop the service.
+     *
+     * @param args Arguments from prunsrv command line
+     **/
+    public static void windowsService(String args[]) {
+        String cmd = "start";
+        if (args.length > 0) {
+            cmd = args[0];
+        }
+
+        if ("start".equals(cmd)) {
+            if(testserverLauncherInstance == null){
+                testserverLauncherInstance = new TestServerMain();
+            }
+
+            testserverLauncherInstance.initCouchbaseLite();
+            testserverLauncherInstance.startServer(true);
+        }
+        else {
+            try{
+                testserverLauncherInstance.stop();
+            }
+            catch (Exception e){
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public void startServer(boolean asService) {
+        NanoHTTPD server = null;
+        try {
+
+
+            File tempDir = new File(TMP_DIR, "TestServerTemp");
+            context = new TestServerContext(tempDir);
+            Server.setContext(context);
+
+            String ip = context.getLocalIpAddress();  //here the context hands out an ip address of 0.0.0.0
+            Server.memory.setIpAddress(ip);
+
+            server = new Server(context, PORT);
+            server.start(5000, false);
+
+            Log.i(TAG, "A HTTP server launched at " + ip + ", port = " + PORT);
+        } catch (IOException var3) {
+            Log.e(TAG, "Couldn't start server:\n" + var3);
+            System.exit(-1);
+        }
+
+        if (asService) {
+            waitForStopService();
+        } else {
+            waitForEnter();
+        }
+
+        if (server != null) {
+            server.stop();
+            Log.i(TAG, "The HTTP Server stopped.\n");
+        }
+    }
+
+    private void waitForEnter() {
+        System.out.println("Server started, Hit Enter to stop.\n");
+        try {
+            System.in.read();
+        } catch (Throwable th) { }
```

Comment:
Please, never, ever do this.  Catch only exceptions that you know how to handle.  ... Here, it appears that you actually don't know how to handle any exceptions at all... I'd suggest, then, not catching them.